www/friendly.html.pm

```diff
@@ -0,0 +1,64 @@
+
+If someone takes issue with something you said or did, resist the urge to be defensive. Just stop doing what it was they complained about and apologize. Even if you feel you were misinterpreted or unfairly accused, chances are good there was something you could’ve communicated better — remember that it’s your responsibility to make your fellow Racketeers comfortable. Everyone wants to get along and we are all here first and foremost because we want to talk about cool technology. You will find that people will be eager to assume good intent and forgive as long as you earn their trust.
+}
+
+◊section{
+Friendly Environment Policy
+
+* We are committed to providing a friendly, safe and welcoming environment for all, regardless of level of experience or personal characteristics.
+* Please avoid using overtly sexual aliases or other nicknames that might detract from a friendly, safe and welcoming environment for all.
+* Please be kind and courteous. There’s no need to be mean or rude.
+* Respect that people have differences of opinion and that every design or implementation choice carries a trade-off and numerous costs.
+* Please keep unstructured critique to a minimum. If you have solid ideas you want to experiment with, make a fork and see how it works.
+* We will exclude you from interaction if you insult, demean or harass anyone. That is not welcome behavior. We interpret the term “harassment” as including the definition in the Citizen Code of Conduct; if you have any lack of clarity about what might be included in that concept, please read their definition. In particular, we don’t tolerate behavior that excludes people in socially marginalized groups.
+* Private harassment is also unacceptable. No matter who you are, if you feel you have been or are being harassed or made uncomfortable by a community member, please contact one of the channel ops or any of the Rust moderation team immediately. Whether you’re a regular contributor or a newcomer, we care about making this community a safe place for you and we’ve got your back.
+* Likewise any spamming, trolling, flaming, baiting or other attention-stealing behavior is not welcome.
+
+Anyone witnessing or subject to unacceptable behavior should ◊link["mailto:modration-team@racket-lang.org"]{EMAIL THE MODERATION TEAM}.
+
+If anynone engages in harassing behavior, the MODERATION TEAM may take any action they deem appropriate, from a warning of the offender to an expulsion.
+} 
+
+◊section{Moderation Policy
+
+These are the policies for upholding our community’s standards of conduct. If you feel that a thread needs moderation, please contact the Rust moderation team .
+
+1. Remarks that violate the Racket standards of conduct, including hateful, hurtful, oppressive, or exclusionary remarks, are not allowed. Cursing is allowed, but never targeting another user, and never in a hateful manner.
```

Comment:
I find this entire policy to be absurd and one-sided towards a very particular perspective of what is offensive and inappropriate behavior. For example, I believe that I am as offended by cursing and blasphemy as other people are by things they might call "hateful, hurtful, oppressive, or exclusionary." Yet, I don't think it is appropriate for me to impose my opinions about what words people should say on the Racket community. Being in the world but not of the world requires me to ignore people when they don't do what I would do and practice charitable forbearance with their behavior. I think the best thing for a Friendly-Environment would be 1 Corinthians 13 and D&C 88:133. I assume that many readers of this policy would find my suggestion absurd; if you do, that's how I feel about this document.

Reply:
@jeapostrophe ~~ I have expressed a similar opinion here (with slightly softer words; see my comments on "user") and in email to @spdegabrielle But we live in times where such a policy is sadly desired by some people and even more sadly necessary to some extent (as the Scala community had to learn thru harsh experiences). 

As for the "cursing" I had actually marked it up and changed my mind. We should delete this sentence.

Reply:
@mfelleisen Sentence removed.

@jeapostrophe I'd like this document to be as universal as possible. Neither 1 Corinthians 13 or D&C 88:133 are absurd, though I had to look both up to my shame. The world needs more love and fellowship. Even our tiny corner. 

I'm hoping the preamble helps a little, but it could obviously be improved. 

> The Racket community aims to improve the world through programming. It started with the goal of introducing everyone to the wonderful world of program design, with a spirit of full inclusion and no emphasis on any specific group.  Over time it has grown into a full-fledged professional community with a well-known reputation for helpfulness and openness on its on-line communication channels. 
> 
> This spirit requires an environment that enables all to participate without fear of personal harassment. We define harassment as unwelcome or hostile behavior, that is, behavior that focuses on people instead of ideas. The organizers consider responses such as “just joking,” or “teasing,” or being “playful” as unacceptable.
> 
> In the Racket community we strive to go the extra step to look out for each other. Don’t just aim to be technically unimpeachable, try to be your best self. In particular, avoid flirting with offensive or sensitive issues, particularly if they’re off-topic; this all too often leads to unnecessary fights, hurt feelings, and damaged trust; worse, it can drive people away from the community entirely.
> 
> If someone takes issue with something you said or did, resist the urge to be defensive. Just stop doing what it was they complained about and apologize. Even if you feel you were misinterpreted or unfairly accused, chances are good there was something you could’ve communicated better — remember that it’s your responsibility to make your fellow Racketeers comfortable. Everyone wants to get along and we are all here first and foremost because we want to talk about cool technology. You will find that people will be eager to assume good intent and forgive as long as you earn their trust.

Reply:
The purpose of a code of conduct is to communicate the intent of the moderation team—it is descriptive, not prescriptive. If the words in a code of conduct do not align with what will actually be enforced, then it is a bad code of conduct, and it is arguably worse than having no code of conduct at all. Therefore, the moderation team should write the code of conduct, if one is provided at all.

Obviously, the presence or absence of a code of conduct (and its contents) can influence which people you attract to your community (whether you like it or not). But that is because people are willing to trust, on some level, that what is contained within the code of conduct reflects the values of the community. If that isn’t true, a code of conduct serves no purpose.

Reply:
thank you.